src/frontend/components/Paster.tsx

```diff
@@ -0,0 +1,46 @@
+import React from "react";
+import { RendererMessenger } from "src/Messaging";
+import { useStore } from "../contexts/StoreContext";
+import { AppToaster } from "./Toaster";
+
+const DEFAULT_FILE_NAME = 'image.png'
+export const Paster = ({ onPaste }: { onPaste?: (data: any) => void; }) => {

```

Comment:
since this function doesn't return anything to render, it'd be more suitable to make it a hook, e.g. `useClipboardImporter` which can be moved to the `src/frontend/hooks` folder